control/mpc_lateral_controller/src/mpc.cpp

```diff
@@ -159,6 +174,10 @@ Float32MultiArrayStamped MPC::generateDiagData(
   append_diag(iteration_num);             // [18] iteration number
   append_diag(runtime);                   // [19] runtime of the latest problem solved
   append_diag(objective_value);           // [20] objective value of the latest problem solved
+  append_diag(std::clamp(
+    Uex(0), -m_steer_lim,
+    m_steer_lim));  // [21] control signal after the saturation constraint (clamp)
+  append_diag(mpc_data_traj_raw.lateral_err);  // [22] lateral error from raw trajectory
```

Comment:
Do you really want to add this variable?

As you may know, `append_diag(mpc_data.lateral_err);              // [5] lateral error` is a similar one to analyze the lateral error. If you add the above variable, it will be very confusing that we have similar two variables.
Therefore, if you want to add the above variable, I highly recommend to organize the variables in the diag and remove them if unnecessary.

Reply:
fixed in d0cfc56e7a210e2590ea5c8864201cc5b0523be8 and dc98701f46db07d0a39892548285848d05f160e1